openmc/capi/filter.py

```diff
@@ -201,6 +213,39 @@ class DistribcellFilter(Filter):
 class EnergyFunctionFilter(Filter):
     filter_type = 'energyfunction'
 
+    def __new__(cls, energy=None, y=None, uid=None, new=True, index=None):
+        return super().__new__(cls, uid=uid, new=new, index=index)
+
+    def __init__(self, energy=None, y=None, uid=None, new=True, index=None):
+        if (energy is None) != (y is None):
+            raise AttributeError("Need both energy and y or neither")
+        super().__init__(uid, new, index)
+        if energy is not None:
+            self.set_interp_data(energy, y)
+
+    def set_interp_data(self, energy, y):
```

Comment:
Let's just call this `set_data` to match the others. Also, need a docstring on this guy.